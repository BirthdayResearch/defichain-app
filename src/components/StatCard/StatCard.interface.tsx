export interface StatCardProps {
  label: string;
  value: string;
  unit: string;
}
export interface StatCardState {}
