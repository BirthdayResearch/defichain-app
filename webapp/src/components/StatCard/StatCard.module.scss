// App and Bootstrap variables required
@import '~bootstrap/scss/functions';
@import '~bootstrap/scss/variables';
@import '~bootstrap/scss/mixins';
@import '../../scss/variables';
@import '../../scss/bootstrap/variables';

.statCard {
  margin-bottom: 24px;
  height: calc(100% - 24px);
}

.cardBody {
  display: inline-block;
  flex-wrap: wrap;
  justify-content: space-between;
  align-items: flex-end;
}

.label {
  font-weight: $font-weight-medium;
  letter-spacing: 0.15px;
  width: 100%;
  margin-bottom: 0.5rem;
}

.valueUnit {
  align-items: flex-end;
}

.value {
  font-weight: $font-weight-semibold;
  font-size: 2.125rem;
  line-height: 2.5rem;
  word-break: break-all;
}

.unit {
  flex-grow: 0;
  text-align: end;
  font-weight: $font-weight-medium;
  letter-spacing: 0.15px;
  color: $secondary;
}

.text {
  text-align: end;
  line-height: 1.5rem;
}

.statCardActionWithIconAnimation {
  svg {
    animation: rotation 1s infinite linear;
    color: $primary;

    &:hover {
      color: $primary;
    }
  }
}

.statCardActionWithIcon {
  flex-grow: 0;
}

@keyframes rotation {
  from {
    transform: rotate(0deg);
    visibility: visible;
  }

  to {
    transform: rotate(359deg);
  }
}
