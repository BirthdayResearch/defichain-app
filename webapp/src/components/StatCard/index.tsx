import BigNumber from 'bignumber.js';
import React from 'react';
import { Card, CardBody, Col, Row } from 'reactstrap';
import NumberMask from '../NumberMask';
import styles from './StatCard.module.scss';

interface StatCardProps {
  label: string;
  value: string;
  unit: string;
  icon?: any;
  refreshFlag?: any;
}

const StatCard: React.FunctionComponent<StatCardProps> = (
  props: StatCardProps
) => {
  return (
    <Card className={styles.statCard}>
      <CardBody className={styles.cardBody}>
        <Row>
          <Col className={styles.label}>{props.label}</Col>
          {props.icon && (
            <Col
              className={`${styles.statCardActionWithIcon} ${
                props.refreshFlag ? styles.statCardActionWithIconAnimation : ''
              }`}
            >
              {props.icon}
            </Col>
          )}
        </Row>
        <Row className={styles.valueUnit}>
          <Col className={styles.value}>
            <NumberMask
              value={new BigNumber(props.value).toFixed(8)}
              defaultValue={0}
            />
          </Col>
          <Col className={`${styles.unit} ${styles.text}`}>{props.unit}</Col>
        </Row>
      </CardBody>
    </Card>
  );
};

export default StatCard;
