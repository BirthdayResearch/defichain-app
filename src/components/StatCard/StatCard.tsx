import React, { Component } from "react";
import { Card, CardBody } from "reactstrap";
import styles from "./StatCard.module.scss";
import { StatCardProps, StatCardState } from "./StatCard.interface";

class StatCard extends Component<StatCardProps, StatCardState> {
  render() {
    return (
      <Card className={styles.statCard}>
        <CardBody className={styles.cardBody}>
          <div className={styles.label}>{this.props.label}</div>
          <div className={styles.value}>{this.props.value}</div>
          <div className={styles.unit}>{this.props.unit}</div>
        </CardBody>
      </Card>
    );
  }
}

export default StatCard;
