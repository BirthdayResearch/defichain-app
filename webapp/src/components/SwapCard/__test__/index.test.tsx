import React from 'react';
import StatCard from '../index';
import { shallow } from 'enzyme';

describe('StatCard component', () => {
  it('should check for snapshot', () => {
    const props = {
      label: 'label',
      dropdownLabel: 'dropdownLabel',
      tokenMap: new Map(),
      name: 1,
      formState: 'formState',
      handleChange: () => {},
      setMaxValue: () => {},
      isLoadingTestPoolSwapTo: true,
      isLoadingTestPoolSwapFrom: true,
      handleDropdown: () => {},
    };
    const wrapper = shallow(<StatCard {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
