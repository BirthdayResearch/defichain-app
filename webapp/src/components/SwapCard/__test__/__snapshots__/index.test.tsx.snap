// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatCard component should check for snapshot 1`] = `
<Card
  tag="div"
>
  <CardBody
    tag="div"
  >
    <div>
      label
    </div>
    <div>
      <div>
        <Input
          disabled={true}
          id="input"
          inputMode="decimal"
          name="amount1"
          onChange={[Function]}
          pattern="[0-9.,]"
          placeholder="0"
          type="text"
        />
      </div>
      <div>
        <SwapDropdown
          dropdownLabel="dropdownLabel"
          formState="formState"
          handleDropdown={[Function]}
          name={1}
          tokenMap={Map {}}
        />
      </div>
    </div>
  </CardBody>
  <CardFooter
    tag="div"
  >
    <div>
      <span>
        Balance
      </span>
      : 
      0
    </div>
    <div>
      <Button
        color="link"
        disabled={true}
        onClick={[Function]}
        size="sm"
        tag="button"
      >
        Max
      </Button>
    </div>
  </CardFooter>
</Card>
`;
